src/osgPlugins/mdl/VVDReader.cpp

```diff
@@ -69,6 +69,9 @@ bool VVDReader::readFile(const std::string & file)
     if (header.magic_number != VVD_MAGIC_NUMBER)
     {
         OSG_NOTICE << "Vertex data file not valid" << std::endl;
+        // Close the file
```

Comment:
I've just done a code review of the surround code and it looks like the original author created a ifstream on the heap inappropriately, and should have just created it on the stack. It looks like the original author misunderstood how new and ifstream constructor work too.  Amazing such old code hadn't been picked up as being sloppy before, I can only guess that almost no one uses this niche plugin.

I have change the code to use a stack created ifstream and checked this in to master.  I'll also merge this with the 3.6 branch.

Reply:
:+1:

Reply:
Huh.......   I just find you are using Coverity  to find potential bugs in this project.   And, Coverity did not find this file resource leak? Coverity may have more powerful feature compared to cppcheck.